apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -210,23 +249,20 @@ public void SetStackTrace(StackTrace stackTrace)
             var currentStack = TraceStack;
             CheckStackNotEmpty(currentStack);
 
-            AnnotateSpan(currentStack.Peek(), TraceLabels.FromStackTrace(stackTrace));
+            AnnotateSpan(currentStack.Peek().TraceSpan, TraceLabels.FromStackTrace(stackTrace));
         }
 
         /// <inheritdoc />
-        public string GetCurrentTraceId()
-        {
-            return _trace.TraceId;
-        }
-
+        public string GetCurrentTraceId() =>_trace.TraceId;
+        
         /// <inheritdoc />
         public ulong? GetCurrentSpanId() => GetCurrentSpanId(TraceStack);
 
         /// <summary>
         /// Gets the current span id of the specified stack or null if none exists.
         /// </summary>
-        private ulong? GetCurrentSpanId(ImmutableStack<TraceSpan> traceStack)
-            => traceStack.IsEmpty ? _rootSpanParentId : traceStack.Peek().SpanId;
+        private ulong? GetCurrentSpanId(ImmutableStack<Span> traceStack) =>
```

Comment:
I think it needs to be changed here and in annotate but when we start a span the parent may already be disposed like in the case when a span starts in a parent after the parent has been disposed.  In the current thread this span will be removed from the stack already but in the child thread it will be just marked as disposed.